src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -911,6 +908,23 @@ public override ValueTask WriteAsync(ReadOnlyMemory<byte> buffer, CancellationTo
 
                     return new ValueTask(http2Stream.SendDataAsync(buffer, cancellationToken));
                 }
+
+                public override Task FlushAsync(CancellationToken cancellationToken)
+                {
+                    if (cancellationToken.IsCancellationRequested)
+                    {
+                        return Task.FromCanceled(cancellationToken);
+                    }
+
+                    Http2Stream http2Stream = _http2Stream;
+
+                    if (http2Stream == null)
+                    {
+                        return Task.CompletedTask;
+                    }
+
+                    return http2Stream._connection.FlushAsync();
```

Comment:
Technically, we only have to flush the connection if we in fact have writes for this stream that have not been flushed yet. 

I don't see a good way to implement this without a lot of hassle though, so it's not worth looking at.

But perhaps we could at least add a comment?

Reply:
> But perhaps we could at least add a comment?

Sure.